### PR TITLE
Allow empty `PauliVSpace` init

### PR DESCRIPTION
**Pull request by Qottmann**
In situations where one wants to iteratively build up a `PaulIVSpace` it is handy to be able to initialize an empty `PauliVSpace`. This is currently not possible and fixed with this PR.

### CODE REVIEW

File: pennylane/pauli/dla/lie_closure.py [hunk 1]
```diff
@@ -249,7 +249,13 @@ def __init__(self, generators, dtype=float, tol=None):
             ]
 
         # Get all Pauli words that are present in at least one Pauli sentence
-        all_pws = list(reduce(set.__or__, [set(ps.keys()) for ps in generators]))
+
+        all_pws = (
+            list(reduce(set.__or__, [set(ps.keys()) for ps in generators]))
+            if len(generators) != 0
+            else []
+        )
```
Comment (dwierichs) [inline pennylane/pauli/dla/lie_closure.py:257]: Given this takes 5 lines now, one could also do this in a "vanilla" `if-else` clause, but that's a matter of taste of course :)

**Review by dwierichs — APPROVED** [review]
Thanks @Qottmann , that's a nice catch!

Would you mind adding your name to the changelog? :)

File: pennylane/pauli/dla/lie_closure.py (hunk 1 above)
Comment (Qottmann) [reply pennylane/pauli/dla/lie_closure.py]: ✅

**Review by trbromley — APPROVED** [review]

**Comment by Alex-Preciado** [comment]
sc-63068

**Pull request merged**